### ISSUE TITLE
Add an option to choose the ASN origin domain

### ISSUE THREAD

**Issue by alarig** [traviscross/mtr#238]
Hi,

If cymru.com is broken, this result in many “AS???” in the output as

alarig@airmure ~ % dig -t TXT 0.0.0.0.2.0.1.0.4.8.8.5.0.0.a.2.origin6.asn.cymru.com.

; <<>> DiG 9.11.2-P1 <<>> -t TXT 0.0.0.0.2.0.1.0.4.8.8.5.0.0.a.2.origin6.asn.cymru.com.
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: SERVFAIL, id: 47639
;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4096
;; QUESTION SECTION:
;0.0.0.0.2.0.1.0.4.8.8.5.0.0.a.2.origin6.asn.cymru.com. IN TXT

;; Query time: 276 msec
;; SERVER: 2a01:cb08:898c:fc00::1#53(2a01:cb08:898c:fc00::1)
;; WHEN: Mon Feb 05 10:06:47 CET 2018
;; MSG SIZE  rcvd: 82

alarig@airmure ~ % mtr -bzw grifon01.ring.nlnog.net
Start: Mon Feb  5 10:08:57 2018
HOST: airmure                                                                                           Loss%   Snt   Last   Avg  Best  Wrst StDev
  1. AS???    2a01cb08898cfc010000000000000001.ipv6.abo.wanadoo.fr (2a01:cb08:898c:fc01::1)              0.0%    10    0.5   0.5   0.4   0.6   0.0
  2. AS???    2a01cb08a00402150193025300760249.ipv6.abo.wanadoo.fr (2a01:cb08:a004:215:193:253           0.0%    10    1.9   1.9   1.4   2.3   0.0
  3. AS???    2a01:cfc4:0:500::b                                                                         0.0%    10    6.7   8.8   6.2  26.1   6.1
  4. AS???    bundle-ether104.auvtr4.aubervilliers.opentransit.net (2a01:cfc4:0:500::7)                 10.0%    10    9.9  10.8   7.3  13.4   1.8
  5. AS???    be5511.rcr21.par05.atlas.cogentco.com (2001:978:3::16d)                                    0.0%    10    7.0   7.1   6.4   8.0   0.3
  6. AS???    be2425.ccr42.par01.atlas.cogentco.com (2001:550:0:1000::8275:3cd)                          0.0%    10    7.8   7.5   7.0   7.9   0.0
  7. AS???    be2108.agr22.par01.atlas.cogentco.com (2001:550:0:1000::8275:3285)                        10.0%    10    7.7   7.7   7.2   9.0   0.4
  8. AS???    te0-0-2-2.rcr11.rns01.atlas.cogentco.com (2001:550:0:1000::9a19:c15)                       0.0%    10   18.5  18.5  17.5  22.6   1.4
  9. AS???    te0-0-2-0.nr11.b015576-0.rns01.atlas.cogentco.com (2001:550:0:1000::9a19:47e)              0.0%    10   18.4  18.4  17.8  19.0   0.0
 10. AS???    2001:978:2:4e::5:2                                                                         0.0%    10   17.0  17.4  16.8  17.9   0.0
 11. AS???    regis.swordarmor.fr (2a00:5884:102::1)                                                     0.0%    10   17.7  17.5  17.0  18.1   0.0
 12. AS???    nlnogring-grifon.swordarmor.fr (2a00:5884:102:1::2)                                        0.0%    10   17.1  17.5  17.1  18.1   0.0

But cymru is not the only service giving this kind of information. It is for example possible to use rezopole.net

alarig@airmure ~ % dig -t TXT 0.0.0.0.2.0.1.0.4.8.8.5.0.0.a.2.origin6.asn.rezopole.net

; <<>> DiG 9.11.2-P1 <<>> -t TXT 0.0.0.0.2.0.1.0.4.8.8.5.0.0.a.2.origin6.asn.rezopole.net
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 54425
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4096
;; QUESTION SECTION:
;0.0.0.0.2.0.1.0.4.8.8.5.0.0.a.2.origin6.asn.rezopole.net. IN TXT

;; ANSWER SECTION:
0.0.0.0.2.0.1.0.4.8.8.5.0.0.a.2.origin6.asn.rezopole.net. 0 IN TXT "204092 | 2a00:5884::/32 | FR | ripe | 2015-08-27 | GRIFON"

;; Query time: 311 msec
;; SERVER: 2a01:cb08:898c:fc00::1#53(2a01:cb08:898c:fc00::1)
;; WHEN: Mon Feb 05 10:07:01 CET 2018
;; MSG SIZE  rcvd: 155

Is it possible to add an option to choose the service we would like to use or something like that?
Thanks.

**Comment by zorun** [comment]
Good idea, see also #221 , especially the last comment by @yvs2014 whose code seems to do almost exactly what you suggest.

I think an even better improvement would be to automatically query several such services, without needing an explicit option.

**Comment by rewolff** [comment]
The problem with that approach is that we don't get any feedback that one of them has stopped until ALL configured services stop working....

**Comment by alarig** [comment]
@rewolff All configured services will never stop working at the same time, so mtr will always display the ASN.

**Comment by rewolff** [comment]
Yes. But just like in a RAID.... you should replace the broken disk as soon as one is broken. But people end up calling me (with my datarecovery hat on) once the second drive fails. Because of the redundancy things keep working untill the last working option is gone. 

By "trying" the services in sequence and having say a 3 second timeout, we'd notice a slowdown once the first service stops working and we maybe get a complaint before all of them stop.... But from my experience, it will easily continue to work until all of them stop and people will accept the slowdown. 

With such a list, it would be best if you cache the "service X worked last time", When you do that people have a one-time slowdown when MTR notices a service goes down and after that things are fast again. But in that ideal situation we'd quickly get the situation that everybody sees once a slowdown as one service shuts down but then everything is quick again until at one point it all stops working. Anyway.... patches accepted.

**Comment by yvs2014** [comment]
@rewolff > closed this as completed in #473

Not sure, if the use of origin[6].the-same-service could resolve the case when that service is off.
Using ipinfo option more-less oftenm, I have finally switched to 'ripewhois' (and kept 'cymru' and several others as an option), but it's a bit bigger piece of code
